explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -44,179 +42,150 @@
-        .thenReturn(StatusCode.CERTIFICATE_NOT_FOUND.get());
-    when(contractExecutionResponseCertificateNotFound.getMessage())
-        .thenReturn("certificate not found");
-    when(contractExecutionResponseContractNotFound.getStatus())
-        .thenReturn(StatusCode.CONTRACT_NOT_FOUND.get());
-    when(contractExecutionResponseContractNotFound.getMessage()).thenReturn("contract not found");
+    when(clientConfig.getCertHolderId()).thenReturn(HOLDER_ID);
   }
 
   @Test
-  public void get_CertificateAndContractNotRegistered_ShouldSucceed() {
+  public void get_ShouldSucceed() {
     // Arrange
-    when(contractExecutionResponseOK.getResult()).thenReturn("{ \"id\": \"bar\" }");
+    JsonObject expected = Json.createObjectBuilder().add("status", StatusCode.OK.get()).build();
+    when(contractExecutionResultOK.getResult()).thenReturn(Optional.of(expected));
     when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseContractNotFound,
-            contractExecutionResponseOK);
+        .thenReturn(contractExecutionResultOK);
 
     // Act
-    JsonObject result = explorer.get(ASSET_ID);
+    JsonObject actual = explorer.get(ASSET_ID);
 
     // Assert
-    verify(clientService, Mockito.times(3))
-        .executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class));
-    assertThat(result.getString("id")).isEqualTo("bar");
+    verify(clientService).executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class));
+    assertThat(actual).isEqualTo(expected);
   }
 
   @Test
   public void get_FailedToRegisterCertificate_ShouldThrowExplorerException() {
     // Arrange
-    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(contractExecutionResponseCertificateNotFound);
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseCertificateNotFound);
-
-    // Act-assert
-    assertThatThrownBy(() -> explorer.get(ASSET_ID))
-        .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
-  }
-
-  @Test
-  public void get_CertificateNotFoundTwice_ShouldThrowExplorerException() {
-    // Arrange
-    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseCertificateNotFound);
+    when(clientService.executeContract(eq( HOLDER_ID + "GET"), any(JsonObject.class)))
+        .thenThrow(clientExceptionCertificateNotFound);
+    doNothing().when(clientService).registerCertificate();
 
     // Act-assert
     assertThatThrownBy(() -> explorer.get(ASSET_ID))
         .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
+        .hasMessage("403 Certificate not found");
```

Comment:
Sorry, the original implementation is not very good and it is a little extension and shouldn't be done in this PR,
but can you update `ExplorerException` to include StatusCode so that it can compare with StatusCode instead of comparing with messages, which is very error-prone ?

Or we can just remove ExplorerException since it is meaningless and just propagate ClientException from client SDK to the client of client SDK.
Maybe the latter might be good.

Reply:
Thank you for the review comment.
Review changes updated in 877c01f76483d2031a19aba97b74a6ad20be66fa